repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -97,13 +100,25 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
   }
 
   /**
-    * Gets the SAMl response from the post encoded body, and decodes it into a plain String.
+    * Gets the SAMl response from the post encoded body, and decodes it.
     *
     * @param request the servlet request that has an encoded body
     * @return the decoded saml response
     * @throws SamlPolicyException if decoding fails
     */
-  def decodeSamlResponse(request: HttpServletRequest): InputStream = ???
+  def decodeSamlResponse(request: HttpServletRequest): InputStream = {
+    try {
+      Option(request.getParameter("SAMLResponse"))
```

Comment:
So this is interesting. `getParameter` actually returns a string, but the content of `SAMLResponse` could be in the body. How does `getParameter` know the encoding of that body? Well, `application/x-www-form-urlencoded` does not actually define the `charset` parameter. In fact, the content is always encoded as `US-ASCII`. To reason about that, consider that the same content could be transmitted over HTTP as a query parameter. Therefore, it's always safe and correct for this method to return a `String` (assuming a correct implementation of `HttpServletRequest`).

Now you might be wondering how exotic characters can be sent using HTML's `application/x-www-form-urlencoded` over HTTP. Well, a `_charset_` parameter can be included as a parameter and can specify the encoding of the content after decoded using `US-ASCII`.